lib/govuk_schemas/assert_matchers.rb

```diff
@@ -0,0 +1,26 @@
+
+require "govuk_schemas/validator"
+
+module GovukSchemas
+  module AssertMatchers
+    def assert_valid_against_publisher_schema(payload, schema_name)
+      validator = GovukSchemas::Validator.new(schema_name, "publisher", payload)
+      assert validator.valid?, validator.error_message
+    end
+
+    def assert_valid_against_links_schema(payload, schema_name)
+      validator = GovukSchemas::Validator.new(schema_name, "links", payload)
+      assert validator.valid?, validator.error_message
+    end
+
+    def assert_valid_against_frontend_schema(payload, schema_name)
+      validator = GovukSchemas::Validator.new(schema_name, "frontend", payload)
+      assert validator.valid?, validator.error_message
+    end
+
+    def assert_valid_against_notification_schema(payload, schema_name)
+      validator = GovukSchemas::Validator.new(schema_name, "notification", payload)
+      assert validator.valid?, validator.error_message
+    end
```

Comment:
Should we have a private method that can handle this duplication?

e.g. 

def assert_valid_against_schema(payload, schema_name, schema_type)
      validator = GovukSchemas::Validator.new(schema_name, schema_type, payload)
      assert validator.valid?, validator.error_message
end